docs/docs/user-guide/training-data-format.mdx

```diff
@@ -0,0 +1,659 @@
+be in the format of the latest version of Rasa Open Source. Training data files with a version greater than is currently
+available for Rasa Open Source will be skipped.
+
+### Example
+
+Here's a short example which keeps all training data in a single file:
+
+```yaml
+version: "2.0"
+
+stories:
+- story: greet and faq
+  steps:
+  - intent: greet
+  - action: utter_greet
+  - intent: faq
+  - action: respond_faq
+
+rules:
+- rule: Greet user
+  steps:
+  - intent: greet
+  - action: utter_greet
+
+responses:
+  faq/language:
+  - text: |
+    I can only do English at the moment
+
+  utter_greet:
+  - text: |
+    Hey there!
+
+nlu:
+- intent: greet
+  examples: |
+  - Hey
+  - Hi
+  - hey there [Sara](name)
+
+- intent: faq/language
+  examples: |
+  - What language do you speak?
+  - Do you only handle english?
+
+e2e_tests:
+- user: |
+    hey
+  intent: greet
+- action: utter_greet
+- user: |
+    what language do you speak
+  intent: faq/language
+- action: respond_faq
+```
+
+
+## NLU Training Data
+
+NLU training data consists of example user utterances categorized by
+**intent**, i.e. what the user is trying to convey or accomplish with their
+message. Training examples can also include **entities**. Entities are structured
+pieces of information that can be extracted from a user's message. The other components
+of NLU training data are there to help the model identify intent and entity
+correctly.
+
+NLU training data is defined under the `nlu` key. Items that can be added under this key are:
+
+- [Training examples](#training-examples) grouped by user intent,
+  optionally with annotated [entities](#entities)
+- [Synonyms](#synonyms)
+- [Regular Expression Features](#regular-expression-features)
+- [Lookup Tables](#lookup-tables)
+
+Here is a simple example of each item type:
+
+```yaml
+nlu:
+- intent: check_balance
+  examples: |
+  - What's my [credit](account) balance?
+  - What's the balance on my [credit card account]{"entity":"account","value":"credit"}
+
+- synonym: credit
+  examples: |
+  - credit card account
+  - credit account
+
+- regex: account_number
+  examples: |
+  - \d{10:12}
+
+- lookup: banks
+  examples: |
+  - JPMC
+  - Comerica
+  - Bank of America
+```
+
+
+### Training Examples
+
+Training examples are grouped by [intent](../glossary#intent) and listed under the
+`examples` key. Usually, you'll list one example per line as follows:
+
+```yaml
+nlu:
+- intent: greet
+  examples: |
+  - hey
+  - hi
+  - whats up
+```
+
+However, its also possible to use an extended format if you want to include
+metadata for your examples:
+
+```yaml
+nlu:
+- intent: greet
+  examples: 
+  - text: |
+      hi
+    metadata:
+      sentiment: neutral
+  - text: |
+      hey there!
+```
+
+The `metadata` key can contain arbitrary key-value data and the parser will not read its value. 
+Metadata can be useful for storing information relevant to an example, but not used by 
```

Comment:
I don't think this is actually true, e.g. the sentiment in the above example would be used in an NLU component (we just don't have one by default).

Reply:
Ok, I was wondering about that. Is this true then?
```
The `metadata` key can contain arbitrary key-value data. The parser will read it's value, but it will only be used if a component in the pipeline needs it. In the example above, the sentiment of 
the example is recorded as metadata. This could be used by a custom component in the pipeline for sentiment analysis.
```

Reply:
yes exactly, that makes more sense. metadata really is more an extension point for people that want to use their own or third party components that need additional data annotations